devlib/module/sched.py

```diff
@@ -0,0 +1,262 @@
+#    Copyright 2018 Arm Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+
+from devlib.module import Module
+from devlib.utils.misc import memoized
+
+from past.builtins import basestring
+
+
+class SchedModule(Module):
+
+    name = 'sched'
+
+    sysfs_root = '/sys/devices/system/'
+
+    @staticmethod
+    def probe(target):
+        has_sd = target.directory_exists(SchedDomainData.sched_domain_root)
+        if not has_sd:
+            return False
+
+        return len(target.list_directory(SchedDomainData.sched_domain_root)) > 0
+
+    def get_sd_info(self, cpu):
```

Comment:
Don't see the need for such a "dummy" wrapper... is it just to avoid passing in target?
I would say clients can directly call `SchedDomainData(self.target, cpu)`, why not?